velox/exec/Window.cpp

```diff
@@ -41,16 +41,84 @@ void initKeyInfo(
   }
 }
 
-void checkDefaultWindowFrame(const core::WindowNode::Function& windowFunction) {
-  VELOX_CHECK_EQ(
-      windowFunction.frame.type, core::WindowNode::WindowType::kRange);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.startType,
-      core::WindowNode::BoundType::kUnboundedPreceding);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.endType, core::WindowNode::BoundType::kCurrentRow);
-  VELOX_CHECK_EQ(windowFunction.frame.startValue, nullptr);
-  VELOX_CHECK_EQ(windowFunction.frame.endValue, nullptr);
+vector_size_t findRowUnboundedPreceding(
+    vector_size_t partitionStartRow,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t /*peerStartRow*/,
+    vector_size_t /*peerEndRow*/,
+    vector_size_t /*currentRow*/) {
+  return partitionStartRow;
+}
+
+vector_size_t findCurrentRowFrameStart(
+    vector_size_t /*partitionStartRow*/,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t peerStartRow,
+    vector_size_t /*peerEndRow*/,
+    vector_size_t /*currentRow*/) {
+  return peerStartRow;
+}
+
+vector_size_t findCurrentRowFrameEnd(
+    vector_size_t /*partitionStartRow*/,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t /*peerStartRow*/,
+    vector_size_t peerEndRow,
+    vector_size_t /*currentRow*/) {
+  return peerEndRow;
+}
+
+vector_size_t findRowUnboundedFollowing(
+    vector_size_t /*partitionStartRow*/,
+    vector_size_t partitionEndRow,
+    vector_size_t /*peerStartRow*/,
+    vector_size_t /*peerEndRow*/,
+    vector_size_t /*currentRow*/) {
+  return partitionEndRow;
+}
+
+const WindowFrameFunctionPtr getStartFrameFunction(
+    core::WindowNode::BoundType startType,
+    core::WindowNode::WindowType type) {
+  switch (startType) {
+    case core::WindowNode::BoundType::kUnboundedPreceding:
+      return findRowUnboundedPreceding;
+    case core::WindowNode::BoundType::kCurrentRow: {
+      if (type == core::WindowNode::WindowType::kRange) {
+        return findCurrentRowFrameStart;
+      } else {
+        VELOX_NYI("Not supported");
+      }
+    }
+    case core::WindowNode::BoundType::kPreceding:
+    case core::WindowNode::BoundType::kFollowing:
+      VELOX_NYI("Not supported");
+    case core::WindowNode::BoundType::kUnboundedFollowing:
+    default:
+      VELOX_FAIL("Invalid frame start value");
```

Comment:
Do we have a check in the WindowNode constructor that frame start is not kUnboundedFollowing? If not, let's add.